### ISSUE TITLE
issue with variable name containing "dummy"?

### ISSUE THREAD

**Issue by adw96** [statdivlab/rigr#98]
Hi @taylorokonek -- do you have any idea what's going on with the following error?

```
library(rigr)
set.seed(1)
dd <- data.frame(y=rnorm(100), x=rnorm(100))
dd$ydummy <- dd$y>0
dd$ybinned <- dd$y>0
regress("mean", ydummy~x, data=dd) # Error in rep(varname, att$dim[2]) : invalid 'times' argument
regress("mean", ybinned~x, data=dd) # no issue
regress("mean", dummy(y>0)~x, data=dd) # probably the ideal way to do this; no issue
```

I appreciate any intel you have on this -- thanks to Ken for letting me know about and putting together the MWE.

**Comment by adw96** [comment]
Digging a bit further -- the issue is in `addArgs()` in `rigr/regress_utils.R` line 787, likely due to `findx <- pmatch(type, c("polynomial", "dummy"))` on line 770. Perhaps we should have warning if any variable name has either of these terms in it?

**Comment by taylorokonek** [comment]
Yes, it would be an issue with "dummy" being included in a variable name here. I think adding a warning would be a good idea. There may be a more elegant solution to finding actual dummy variables (other than pmatch) which could also work as a more permanent solution, but may require a bit more time/effort to code

**Comment by yiqunchen** [comment]
Potential fix in [3e035bc](https://github.com/statdivlab/rigr/commit/3e035bc2e7d89e95cdacb9847d3d2291d1028009) -- I changed the pattern matching to dummy( and polynomial( instead of dummy and polynomial -- now unless users name their variables with a single open parenthesis (hopefully obscure??), `regress` should be able to handle it.

**Comment by adw96** [comment]
brilliant!
